common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -208,15 +214,17 @@ public void run() {
                         finalFuture.setException(new ExecutionException(t));
                     } finally {
                         synchronized (mapAccessLock) {
-                            final FinalizableResultFuture mapFuture = sExecutingCommandMap.remove(command);
-                            if (mapFuture == null) {
-                                // If this has happened, the command that we started with has mutated.  We will
-                                // examine every entry in the map, find the one with the same object identity
-                                // and remove it.
-                                // ADO:TODO:1153495 - Rekey this map with stable string keys.
-                                Logger.error(TAG, "The command in the map has mutated " + command.getClass().getCanonicalName()
-                                        + " the calling application was " + command.getParameters().getApplicationName(), null);
-                                cleanMap(command);
+                            if (command.isEligibleForCaching()) {
```

Comment:
We probably need to alter the CommandDispatcherTest so that the TestCommand used to verify that this works is eligible for caching.

Reply:
And there's no need to be synchronized for this check.